QueryToSocketDistributor.h incompatible with Boost 1.89 due to deadline_timer

I'm trying to compile current git (https://github.com/ad-freiburg/qlever/commit/e92d9d18c) against boost-libs 1.89.0:
```
FAILED: src/util/http/CMakeFiles/http.dir/websocket/QueryToSocketDistributor.cpp.o 
/usr/bin/c++ -DBOOST_ASIO_DISABLE_AWAITABLE_FRAME_RECYCLING -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_IOSTREAMS_NO_LIB -DBOOST_RANDOM_DYN_LINK -DBOOST_RANDOM_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_URL_DYN_LINK -DBOOST_URL_NO_LIB -DLOGLEVEL=INFO -DQUERY_CANCELLATION_MODE=ENABLED -DRANGE_V3_COMBINE_WITH_STD -DSPATIALJOIN_NO_BZIP2=True -DSPATIALJOIN_NO_ZLIB=True -I./build/_deps/ctre-src/single-header -I./build/_deps/fsst-src -I./build/_deps/range-v3-src/include -I./src -I./build/_deps/antlr-src/runtime/Cpp/runtime/src/atn -I./build/_deps/antlr-src/runtime/Cpp/runtime/src/dfa -I./build/_deps/antlr-src/runtime/Cpp/runtime/src/internal -I./build/_deps/antlr-src/runtime/Cpp/runtime/src/misc -I./build/_deps/antlr-src/runtime/Cpp/runtime/src/support -I./build/_deps/antlr-src/runtime/Cpp/runtime/src/tree -I./build/_deps/antlr-src/runtime/Cpp/runtime/src/tree/pattern -I./build/_deps/antlr-src/runtime/Cpp/runtime/src/tree/xpath -I./build/_deps/nlohmann-json-src/include -I./build/_deps/re2-src -I./build/_deps/spatialjoin-src/src/util/.. -I./build/_deps/spatialjoin-src/src -isystem ./build/_deps/antlr-src/runtime/Cpp/runtime/src -isystem ./build/_deps/abseil-src -isystem ./build/_deps/googletest-src/googletest/include -isystem ./build/_deps/googletest-src/googletest -isystem ./build/_deps/googletest-src/googlemock/include -isystem ./build/_deps/googletest-src/googlemock -isystem ./build/_deps/s2-src/src -march=native -mtune=native -O1 -pipe -fpermissive -fno-plt -fexceptions -Wformat -Wno-stringop-overflow -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-error=incompatible-pointer-types -fPIC -Wp,-D_GLIBCXX_ASSERTIONS -Wno-zero-as-null-pointer-constant -fPIC  -Wall -Wextra  -O3 -DNDEBUG -O3 -std=gnu++20 -fcoroutines -fdiagnostics-color=always -Wno-attributes -MD -MT src/util/http/CMakeFiles/http.dir/websocket/QueryToSocketDistributor.cpp.o -MF src/util/http/CMakeFiles/http.dir/websocket/QueryToSocketDistributor.cpp.o.d -o src/util/http/CMakeFiles/http.dir/websocket/QueryToSocketDistributor.cpp.o -c ./src/util/http/websocket/QueryToSocketDistributor.cpp
In file included from ./src/util/http/websocket/QueryToSocketDistributor.cpp:5:
./src/util/http/websocket/QueryToSocketDistributor.h:29:16: error: 'deadline_timer' in namespace 'ad_utility::websocket::net' does not name a type
   29 |   mutable net::deadline_timer infiniteTimer_;
      |                ^~~~~~~~~~~~~~
./src/util/http/websocket/QueryToSocketDistributor.cpp: In constructor 'ad_utility::websocket::QueryToSocketDistributor::QueryToSocketDistributor(boost::asio::io_context&, const std::function<void(bool)>&)':
./src/util/http/websocket/QueryToSocketDistributor.cpp:15:7: error: class 'ad_utility::websocket::QueryToSocketDistributor' does not have any field named 'infiniteTimer_'
   15 |       infiniteTimer_{strand_, static_cast<net::deadline_timer::time_type>(
      |       ^~~~~~~~~~~~~~
./src/util/http/websocket/QueryToSocketDistributor.cpp:15:48: error: 'deadline_timer' in namespace 'ad_utility::websocket::net' does not name a type
   15 |       infiniteTimer_{strand_, static_cast<net::deadline_timer::time_type>(
      |                                                ^~~~~~~~~~~~~~
./src/util/http/websocket/QueryToSocketDistributor.cpp:15:62: error: expected '>' before '::' token
   15 |       infiniteTimer_{strand_, static_cast<net::deadline_timer::time_type>(
      |                                                              ^~
./src/util/http/websocket/QueryToSocketDistributor.cpp:16:54: error: 'pos_infin' is not a member of 'boost::posix_time'
   16 |                                   boost::posix_time::pos_infin)},
      |                                                      ^~~~~~~~~
./src/util/http/websocket/QueryToSocketDistributor.cpp: In member function 'boost::asio::awaitable<void> ad_utility::websocket::QueryToSocketDistributor::waitForUpdate() const':
./src/util/http/websocket/QueryToSocketDistributor.cpp:24:27: error: 'infiniteTimer_' was not declared in this scope
   24 |   auto [error] = co_await infiniteTimer_.async_wait(
      |                           ^~~~~~~~~~~~~~
./src/util/http/websocket/QueryToSocketDistributor.cpp: In member function 'void ad_utility::websocket::QueryToSocketDistributor::wakeUpWaitingListeners()':
./src/util/http/websocket/QueryToSocketDistributor.cpp:39:3: error: 'infiniteTimer_' was not declared in this scope
   39 |   infiniteTimer_.expires_at(boost::posix_time::pos_infin);
      |   ^~~~~~~~~~~~~~
./src/util/http/websocket/QueryToSocketDistributor.cpp:39:48: error: 'pos_infin' is not a member of 'boost::posix_time'
   39 |   infiniteTimer_.expires_at(boost::posix_time::pos_infin);
      |                                                ^~~~~~~~~
```

--- a/src/util/http/websocket/QueryToSocketDistributor.cpp
+++ b/src/util/http/websocket/QueryToSocketDistributor.cpp
@@ -12,8 +12,7 @@ namespace ad_utility::websocket {
 QueryToSocketDistributor::QueryToSocketDistributor(
     net::io_context& ioContext, const std::function<void(bool)>& cleanupCall)
     : strand_{net::make_strand(ioContext)},
-      infiniteTimer_{strand_, static_cast<net::deadline_timer::time_type>(
-                                  boost::posix_time::pos_infin)},
+      infiniteTimer_{strand_, net::steady_timer::time_point::max()},
       cleanupCall_{
           cleanupCall,
           [](const auto& cleanupCall) { std::invoke(cleanupCall, false); }},
@@ -36,7 +35,7 @@ net::awaitable<void> QueryToSocketDistributor::waitForUpdate() const {
 
 void QueryToSocketDistributor::wakeUpWaitingListeners() {
   // Setting the time anew automatically cancels waiting operations
-  infiniteTimer_.expires_at(boost::posix_time::pos_infin);
+  infiniteTimer_.expires_at(net::steady_timer::time_point::max());
 }
 
 // _____________________________________________________________________________

--- a/src/util/http/websocket/QueryToSocketDistributor.h
+++ b/src/util/http/websocket/QueryToSocketDistributor.h
@@ -26,7 +26,7 @@ class QueryToSocketDistributor
     : public std::enable_shared_from_this<QueryToSocketDistributor> {
   /// Strand to synchronize all operations on this class
   net::strand<net::any_io_executor> strand_;
-  mutable net::deadline_timer infiniteTimer_;
+  mutable net::steady_timer infiniteTimer_;
   /// Vector that stores the actual data, so all websockets can read it at
   /// their own pace.
   std::vector<std::shared_ptr<const std::string>> data_{};

--- a/test/MessageSenderTest.cpp
+++ b/test/MessageSenderTest.cpp
@@ -6,6 +6,7 @@
 #include <gtest/gtest.h>
 
 #include <boost/asio/experimental/awaitable_operators.hpp>
+#include <chrono>
 
 #include "util/AsyncTestHelpers.h"
 #include "util/http/websocket/MessageSender.h"
@@ -19,6 +20,7 @@ using ad_utility::websocket::QueryRegistry;
 
 using namespace boost::asio::experimental::awaitable_operators;
 using namespace std::string_literals;
+using namespace std::chrono_literals;
 
 using ::testing::Pointee;
 using ::testing::VariantWith;
@@ -36,7 +38,7 @@ ASYNC_TEST(MessageSender, destructorCallsSignalEnd) {
   { MessageSender dummy{std::move(queryId), queryHub}; }
 
   auto impl = [&]() -> net::awaitable<void> {
-    net::deadline_timer timer{ioContext, boost::posix_time::seconds(2)};
+    net::steady_timer timer{ioContext, 2s};
 
     auto result = co_await (distributor->waitForNextDataPiece(0) ||
                             timer.async_wait(net::use_awaitable));
@@ -65,7 +67,7 @@ ASYNC_TEST(MessageSender, callingOperatorBroadcastsPayload) {
     updateWrapper("Still");
     updateWrapper("Dre");
 
-    net::deadline_timer timer{ioContext, boost::posix_time::seconds(2)};
+    net::steady_timer timer{ioContext, 2s};
 
     auto impl = [&]() -> net::awaitable<void> {
       auto result = co_await (distributor->waitForNextDataPiece(0) ||

